spec/function.dd

```diff
@@ -1234,20 +1234,42 @@ int foo(in int x, out int y, ref int z, int q);
     implemented so it's current implementation is equivalent to $(D const).  It is recommended
     to avoid using $(D in) until it is properly defined and implemented.  Use $(D scope const)
     or $(D const) explicitly instead.)
-    $(TROW $(D out), parameter is initialized upon function entry with the default value
+    $(TROW $(D ref), parameter is passed by reference)
+    $(TROW $(D out), parameter is passed by reference and initialized upon function entry with the default value
     for its type)
 
-    $(TROW $(D ref),   parameter is passed by reference)
-    $(TROW $(D scope), references in the parameter
+    $(TROW $(D scope), references to the parameter
     cannot be escaped (e.g. assigned to a global variable).
-    Ignored for parameters with no references)
+    Ignored for parameters with no references).
     $(TROW $(D lazy), argument is evaluated by the called function and not by the caller)
         $(TROW $(D const), argument is implicitly converted to a const type)
     $(TROW $(D immutable), argument is implicitly converted to an immutable type)
     $(TROW $(D shared), argument is implicitly converted to a shared type)
     $(TROW $(D inout), argument is implicitly converted to an inout type)
     )
 
+$(H3 $(LNAME2 ref-params, Ref and Out Parameters))
+
+    $(P By default, parameters take rvalue arguments.
+    A Ref parameter takes an lvalue argument, so changes to its value will operate
+    on the caller's argument.)
+    
+$(SPEC_RUNNABLE_EXAMPLE_COMPILE
```

Comment:
Why not run + assert?

Reply:
I just re-ordered existing code. Assert would be better, but this file doesn't seem to use it.

Reply:
@wilzbach Changed to use assert.